### PR TITLE
fix: Fixes to make Megatron backend match dtensor

### PR DESCRIPTION
**Pull request by ashors1**
# What does this PR do ?

**Add a one line overview of what this PR aims to accomplish.**

# Issues
List issues that this PR closes ([syntax](https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)):

# Usage
* **You can potentially add a usage example below**

```python
# Add a code snippet demonstrating how to use this
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](/NVIDIA-NeMo/RL/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you run the unit tests and functional tests locally? Visit our [Testing Guide](/NVIDIA-NeMo/RL/blob/main/docs/testing.md) for how to run tests
- [ ] Did you add or update any necessary documentation? Visit our [Document Development Guide](/NVIDIA-NeMo/RL/blob/main/docs/documentation.md) for how to write, build and test the docs.

# Additional Information
* ...

## Summary by CodeRabbit

* **New Features**
  * Added supervised fine-tuning recipe configuration for Qwen2.5-Math-7B with Megatron tensor parallelism support.

* **Tests**
  * Added gradient norm consistency validation tests across different parallelism configurations.

* **Chores**
  * Simplified configuration format by removing deprecated distributed data parallel settings from multiple configuration files.
  * Enhanced internal gradient handling for per-token loss computation in policy training.

### CODE REVIEW

File: nemo_rl/models/policy/megatron_policy_worker.py [hunk 1]
```diff
@@ -683,9 +686,7 @@ def __init__(
                 overlap_param_gather=self.cfg["megatron_cfg"][
                     "distributed_data_parallel_config"
                 ]["overlap_param_gather"],
-                average_in_collective=self.cfg["megatron_cfg"][
-                    "distributed_data_parallel_config"
-                ]["average_in_collective"],
+                average_in_collective=False, # average in collective is not supported with per-token loss
```
Comment (terrykong) [inline nemo_rl/models/policy/megatron_policy_worker.py]: Could you add a descriptive comment that explains why this is false and if it should ever be true? This bug wasn't trivial to find, so I want to make sure we capture your findings

File: nemo_rl/models/policy/megatron_policy_worker.py [hunk 2]
```diff
@@ -658,6 +658,9 @@ def __init__(
             "https://github.com/NVIDIA-NeMo/RL/blob/bccbc377705a81a1f4b3c31ad9767bcc15f735a8/nemo_rl/algorithms/sft.py#L175-L179."
         )
 
+        ## TODO: make sure this works with sequence-level losses as well
```
Comment (terrykong) [inline nemo_rl/models/policy/megatron_policy_worker.py]: Same comment about descriptive comment on why this is set the way it is

also, if sequence level losses aren't fixed, does that mean DPO doesn't work, or simply not tested?

File: nemo_rl/models/policy/megatron_policy_worker.py [hunk 3]
```diff
@@ -658,6 +658,21 @@ def __init__(
...
+        ## These settings are required for correct gradient computations in mcore
+        ## when calculate_per_token_loss is True, there is no scaling of the gradient in mcore,
+        ## so we handle the scaling in nemo-rl.
+        ## perform_initialization = True is a workaround to ensure the correct tensor parallel attributes are set
+        ## on the TP-sharded parameters.
+        model_cfg.calculate_per_token_loss = True
+        model_cfg.perform_initialization = False
+
+        assert (
+            "aux_loss" not in self.model_cfg.moe_router_load_balancing_type
```
Comment (ahmadki) [inline nemo_rl/models/policy/megatron_policy_worker.py]: `MegatronPolicyWorker` doesn't have a  `model_cfg` member, `self.model_cfg` should be `model_cfg`

Comment (ashors1) [reply nemo_rl/models/policy/megatron_policy_worker.py]: Done, thanks for the catch

**Review by terrykong — APPROVED** [review]

**Review by ahmadki — APPROVED** [review]

**Review by terrykong — APPROVED** [review]

**Review by terrykong — DISMISSED** [review]
